### DISCUSSION TITLE
Expo support for new react native architecture

### DISCUSSION THREAD

**Discussion by AlexanderEggers** [expo/expo#16528] [Archived: Questions] [answered]
Are there any expo related plans to support the new react native architecture which is going to be introduced with 0.68?

**Comment by barthap** [comment, marked as answer]
Yeah, there are such plans (and even steps made in this direction) to directly use JSI in [Expo Modules](https://docs.expo.dev/modules/overview/) infrastructure. It was mentioned in [this blog post](https://blog.expo.dev/a-peek-into-the-upcoming-sweet-expo-module-api-6de6b9aca492) - see the _"JSI Integration"_ paragraph.

**Reply by raajnadar** [reply]
I have a very small app only 2-3 pages, I just want to test the new arch is that possible to use it with Expo Go on managed or bare? On expo SDK 46

**Reply by keith-kurak** [reply]
Expo 45/46 do allow you to try out the new architecture (with caveats), but with development client/ standalone build only (no Expo Go). See notes on the [latest SDK release here](https://blog.expo.dev/expo-sdk-46-c2a1655f63f7):

> While we’re excited about React Native’s New Architecture (alongside React 18 concurrent features), we can’t recommend it yet to any but the most adventurous developers: we don’t support it on Expo module view managers yet, and most libraries in the ecosystem don’t either.

**Reply by raajnadar** [reply]
> we don’t support it on Expo module view managers yet, and most libraries in the ecosystem don’t either.

Is there any way to know which expo modules can be used with the new architecture, and which cannot be used?

**Reply by metkm** [reply]
I still don't understand how we enable new architecture with expo. Any tutorials? It looks like the blog posts doesn't say anything about how we use it.

**Reply by evelant** [reply]
Yeah it's unclear how to enable it on expo. Seemingly not in the docs anywhere. There's the `experiments -> turboModules` config option in app.json but I'm suspecting that only enables turbomodules, not fabric.

**Reply by Lauro235** [reply]
Have any of you found this error and do you know how much impact it has on the development process.?

**Cannot resolve symbol 'BuildConfig'...** 

I found the generated BuildConfig file...
```
public final class BuildConfig {
  public static final boolean DEBUG = Boolean.parseBoolean("true");
  public static final String APPLICATION_ID = "com.dicerollingapp";
  public static final String BUILD_TYPE = "debug";
  public static final int VERSION_CODE = 1;
  public static final String VERSION_NAME = "1.0";
  // Field from default config.
  public static final boolean IS_NEW_ARCHITECTURE_ENABLED = false;
}
```

I'm guessing this is what is causing the error above? I've gone to town with StackOverflow and Github, but as this is an expo host react-native app, I'm not sure how much I can do here?

**Reply by iegik** [reply]
How to add here also `  public static final boolean IS_HERMES_ENABLED = false;`?
